demos/common/samples/args_helper.hpp

```diff
@@ -26,6 +26,17 @@
 #include <dirent.h>
 #endif
 
+#define DEFINE_INPUT_FLAGS \
```

Comment:
I see no benefits from using a macro, instead of constant inside header.
In fact,  arguments separation between different headers seems clumsy. I suggest you to think more object-oriented way in that case.

Reply:
> I see no benefits from using a macro, instead of constant inside header.

I agree WRT the message constants, but not the flag definitions. As a rule of thumb, a header should not define any objects that aren't `static const`, because doing so risks either multiple definition errors (if the object isn't `static`) or multiple copies of the object that can unexpectedly desynchronize (if it isn't `const`) if the header is included multiple times.

(For C++ 17 and up, I would update that to "no objects that aren't `inline const`", but that's not very relevant for now. 🙂)

Reply:
@Wovchena So what about this? I still think you should bring the constants out of the macro.

Reply:
I think I don't follow.
>  a header should not define any objects that aren't `static const`

is the reason why a macro with `DEFINE_string` and `DEFINE_bool` inside is needed. And
> I see no benefits from using a macro, instead of constant inside header.

is the reason why `input_message` and `loop_message` shouldn't be in the macro.

But
> In fact, arguments separation between different headers seems clumsy.

suggests that I should keep vars close to each other. Why I can't go further and put `input_message` and `loop_message` as close as possible to the flags they correspond to. Into the macro. So having `input_message` and `loop_message` with `DEFINE_string` and `DEFINE_bool` is the benefit "from using a macro, instead of constant inside header".

Reply:
Moving the message constants out of the macro doesn't make them much less close to the flag definitions. You'd essentially just rearrange the lines in `args_helper.hpp` a little bit.

The benefits from doing this are minor, but they do exist. For instance, if you use "go to definition" on `input_message`, it will bring you do the actual definition, and not the macro invocation.

Reply:
Updated